### ISSUE TITLE
`i18n-tasks missing` resulting in false positive when `I18n.t` is called from a helper

### ISSUE THREAD

**Issue by lucascaton** [glebm/i18n-tasks#191]
Hi there, I think have found a bug :-)

With a simple Rails app:

```ruby
# app/helpers/pages_helper.rb
module PagesHelper
  def locale_test
    t '.something'
  end
end
```

```erb
<%# app/views/pages/index.html.erb %>
<h1><%= locale_test %></h1>
```

```yaml
# config/locales/en.yml
en:
  pages:
    index:
      something: 'Something!!!'
```

It works:
![image](https://cloud.githubusercontent.com/assets/86755/13547118/48fd1d94-e310-11e5-84a6-d06e93727463.png)

But:

```
$ i18n-tasks missing

Missing translations (1) | i18n-tasks v0.9.4
+--------+------------------------------------+----------------------------------+
| Locale | Key                                | Value in other locales or source |
+--------+------------------------------------+----------------------------------+
|  all   | pages_helper.locale_test.something | app/helpers/pages_helper.rb:3    |
+--------+------------------------------------+----------------------------------+
```

**Comment by lucascaton** [comment]
I tried to add `# i18n-tasks-use`, but it didn't work as well:

```ruby
# app/helpers/pages_helper.rb
module PagesHelper
  def locale_test
    # i18n-tasks-use t('pages.index.something')
    t '.something'
  end
end
```

(running `$ i18n-tasks missing` I get the same false positive)

**Comment by glebm** [comment]
Unfortunately, this is not a bug but a limitation of i18n-tasks. The static analysis always resolves relative keys based on the filename (+ the enclosing method name in case of .rb files).

There are ways to work around this. First, make i18n-tasks ignore this key:

```yaml
ignore_missing:
- pages_helper.locale_test.something
```

Then, there are two options:

A) Add `i18n-tasks-use` hints at all call sites, e.g.:
    
```html.erb
<% # i18n-tasks-use t('pages.index.something') %>
<h1><%= locale_test %></h1>
```

This is a good option if there are only a few call sites.

B) Add a custom scanner like this:

```ruby
# config/i18n-tasks/something_scanner.rb
require 'i18n/tasks/scanners/file_scanner'
class SomethingScanner < I18n::Tasks::Scanners::FileScanner
  include I18n::Tasks::Scanners::RelativeKeys
  include I18n::Tasks::Scanners::OccurrenceFromPosition

  # @return [Array<[absolute key, Results::Occurrence]>]
  def scan_file(path)
    text = read_file(path)
    text.scan(/^\s*<%=\s*locale_test\b/).map do |_match|
      occurrence = occurrence_from_position(
          path, text, Regexp.last_match.offset(0).first)
      [absolute_key('.something', path), occurrence]
    end
  end
end
I18n::Tasks.add_scanner 'SomethingScanner', only: %w(*.html.erb)
```

```yaml
# config/i18n-tasks.yml.erb
<% require './config/i18n-tasks/something_scanne.rb' %>
```

This is a better option if there are many call sites.

**Comment by lucascaton** [comment]
Hi @glebm, thanks for replying so quickly.

Wouldn't it be worth adding somehow this `Scanner` class to the gem codebase?

**Comment by glebm** [comment]
Perhaps it would make sense to add a scanner that maps patterns to keys, so it can be configured with something like:

```yaml
<% I18n::Tasks.add_scanner, 'I18n::Tasks::Scanners::PatternMapper', only: %w(*.html.erb),
  patterns: [[/^\s*<%=\s*locale_test\b/, '.something']] %>
```

**Comment by glebm** [comment]
Released in v0.9.5. Simply add this to the config:

```erb
<%= I18n::Tasks.add_scanner 'I18n::Tasks::Scanners::PatternMapper',
        only: %w(*.html.erb),
        patterns: [['<%=\s*locale_test\b', '.something']] %>
```

**Comment by lucascaton** [comment]
Thanks @glebm@
